marpa/lib/Guacamole.pm

```diff
@@ -976,29 +1006,98 @@ OpFile ::=
     | OpFileAccessTime
     | OpFileChangeTime
 
-QLikeExpression ::= QLikeExpressionExpr
-
-QLikeExpressionExpr ~ QLikeFunction '(' NonRParenOrEscapedParens_Many               ')'
-                    | QLikeFunction '{' NonRBraceOrEscapedBraces_Many               '}'
-                    | QLikeFunction '<' NonRAngleOrEscapedAngles_Many               '>'
-                    | QLikeFunction '[' NonRBracketOrEscapedBrackets_Many           ']'
-                    | QLikeFunction '/' NonForwardSlashOrEscapedForwardSlashes_Many '/'
-                    | QLikeFunction '!' NonExclamPointOrEscapedExclamPoints_Many    '!'
-                    | QLikeFunction '()'
-                    | QLikeFunction '{}'
-                    | QLikeFunction '<>'
-                    | QLikeFunction '[]'
-                    | QLikeFunction '//'
-                    | QLikeFunction '!!'
+QLikeValue ::= QLikeValueExpr | QLikeValueExprWithMods
+
+QLikeValueExpr
+    ~ QLikeFunction '(' NonRParenOrEscapedParens_Many               ')'
+    | QLikeFunction '{' NonRBraceOrEscapedBraces_Many               '}'
+    | QLikeFunction '<' NonRAngleOrEscapedAngles_Many               '>'
+    | QLikeFunction '[' NonRBracketOrEscapedBrackets_Many           ']'
+    | QLikeFunction '/' NonForwardSlashOrEscapedForwardSlashes_Many '/'
+    | QLikeFunction '!' NonExclamPointOrEscapedExclamPoints_Many    '!'
+    | QLikeFunction '()'
+    | QLikeFunction '{}'
+    | QLikeFunction '<>'
+    | QLikeFunction '[]'
+    | QLikeFunction '//'
+    | QLikeFunction '!!'
 
 QLikeFunction ~ OpKeywordQ
               | OpKeywordQq
               | OpKeywordQx
               | OpKeywordQw
-              | OpKeywordQr
+
+# Here we begin with "qr//" and "m//" which can have parameters
+# Then we continue with "s///", "tr///", and "y///" which have two args, not one
+# "//" follow at the end
+# This is really ugly :/ because I couldn't use NonRParenOrEscapedParens_Many*
```

Comment:
Can `NonRParenOrEscapedParens_Many` use a star instead of a plus, to make itself optional?

Reply:
I tried and it didn't work. :(

Reply:
That's also why I put the comment :/